gui/src/OSNProgressDisplay.cpp

```diff
@@ -0,0 +1,103 @@
+/********************************************************************************
+* Copyright 2017 DigitalGlobe, Inc.
+* Author: Kevin McGee
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+********************************************************************************/
+
+#include "OSNProgressDisplay.h"
+#include "utility/Error.h"
+
+#include <algorithm>
+#include <chrono>
+#include <cmath>
+#include <cstdio>
+#include <iostream>
+
+using std::chrono::milliseconds;
+using std::cv_status;
+using std::defer_lock;
+using std::lock_guard;
+using std::mutex;
+using std::string;
+using std::this_thread::sleep_for;
+using std::thread;
+using std::unique_lock;
+using std::vector;
+
+OSNProgressDisplay::OSNProgressDisplay(const dg::deepcore::ProgressCategories &categories, int intervalMs) :
+        categories_(categories),
+        progress_(categories.size(), 0.0F),
+        oldProgress_(categories.size(), -1.0F),
+        interval_(intervalMs)
+{
+    DG_CHECK(categories_.size(), "No categories specified");
+}
+
+OSNProgressDisplay::~OSNProgressDisplay()
+{
+    stop();
+}
+
+void OSNProgressDisplay::start()
+{
+    stop();
+
+    thread_ = std::thread([this](){
```

Comment:
I meant that I am not in favor of the global using-declarations in the first place such as  `using std::vector`, `using boost::make_unique`, `using cv::Size`, etc.  I think we should explicitly type `std::vector` for instance since it is only five additional characters and makes it explicit what library we are referencing.  So if I am reading code and I see `waitKey(0)` I do not have to scroll back to the top to see for instance `using cv::waitKey` to know that the function is from OpenCV.  As for the length of nested namespaces we can use [`namespace aliases`](http://en.cppreference.com/w/cpp/language/namespace_alias) like in python: `namespace ddi = dg::deepcore::imagery` or `namespace imagery = dg::deepcore::imagery`